calibr8/core.py

```diff
@@ -51,52 +51,111 @@ def __init__(
         hdi_prob : float
             Probability mass in the HDI
         """
-        self.median = median
-        self.eti_x = eti_x
-        self.eti_pdf = eti_pdf
-        self.eti_prob = eti_prob
-        self.hdi_x = hdi_x
-        self.hdi_pdf = hdi_pdf
-        self.hdi_prob = hdi_prob
+        self._eti_x = eti_x
+        self._eti_pdf = eti_pdf
+        self._eti_prob = eti_prob
+        self._hdi_x = hdi_x
+        self._hdi_pdf = hdi_pdf
+        self._hdi_prob = hdi_prob
 
-    def __repr__(self) -> str:
-        result = (
-            str(type(self))
-            + f"\n    ETI ({round(self.eti_prob, 3) * 100:.1f} %): [{round(self.eti_lower, 4)}, {round(self.eti_upper, 4)}] Δ={round(self.eti_width, 4)}"
-            + f"\n    HDI ({round(self.hdi_prob, 3) * 100:.1f} %): [{round(self.hdi_lower, 4)}, {round(self.hdi_upper, 4)}] Δ={round(self.hdi_width, 4)}"
-        )
-        return result
+    @property
+    def eti_x(self) -> numpy.ndarray:
+        """Values of the independent variable in the interval [eti_lower, eti_upper]"""
+        return self._eti_x
 
     @property
-    def eti_lower(self) -> float:
+    def eti_pdf(self) -> numpy.ndarray:
+        """Values of the posterior probability density at the positions `eti_x`."""
+        return self._eti_pdf
+
+    @property
+    def eti_lower(self) -> Union[float, numpy.ndarray]:
         """ Lower bound of the ETI. This is the first value in `eti_x`. """
-        return self.eti_x[0]
+        return self._eti_x[..., 0]
```

Comment:
Why do you need the ellipsis object here?

Reply:
The array can also be 2-dimensional. For example with a trivariate calibration model:
```
array([[0.11, 0.62],    # concentration
       [0.04, 0.25],    # pH
       [0.58, 0.94]])   # temperature
```
The `arr[..., 0]` with the above array then selects the 0ths entry of the last dimension, so `[0.11, 0.04, 0.58]`.

The same slicing applied to a `(2,)` shaped array such as `[0.1, 0.2]` also gives the 0ths entry of the last dimension (0.1).

In contrast, the slicing without the `Ellipsis` would select the first _row_ of the 2-dimensional array.

*tl;dr:* With the ellipsis-based slicing we can keep ETI/HDI arrays as two-column matrices.

Reply:
Ah, I see.` [:, 0]` would do the same in the 2D case, but would fail in 1D?

Reply:
Correct